contracts/rewards/src/state.rs

```diff
@@ -200,7 +200,7 @@ impl RewardsPool {
         if self.balance < reward {
             return Err(ContractError::PoolBalanceInsufficient.into());
         }
-        self.balance -= reward;
+        self.balance = self.balance.saturating_sub(reward);
```

Comment:
```
self.balance = self.balance.checked_sub(reward).map_err(|_| ContractError::PoolBalanceInsufficient.into())
```

Reply:
yeah, at first I was thinking that it doesn't really matter since there was a check before, but this actually can replace that previous check